main.py

```
@@ -139,29 +176,163 @@
     return {"message": "Welcome to the Trady AI Backend"}
 
 @app.post("/api/upload", tags=["Portfolio"])
-async def upload_portfolio(file: UploadFile = File(...)):
+async def upload_portfolio(file: UploadFile = File(...)) -> dict:
```

Ruff (0.11.9)

Redefinition of unused `upload_portfolio` from line 81

(F811)